python/ml4ir/base/model/scoring/interaction_model.py

```diff
@@ -9,93 +9,122 @@
 
         Parameters
         ----------
-        inputs : dict
-            Dictionary of the inputs to the tensorflow keras model
-
-        Returns
-        -------
-        train_features : `tf.Tensor`
-            Dense tensor object that is used for training
-        metadata_features : dict
-            Dictionary of feature tensors that can be used for
-            computing custom metrics and losses
+        feature_config : `FeatureConfig` object
+            FeatureConfig object that defines list of model features
+            and the feature transformation functions to be used on each
+        tfrecord_type : {"example", "sequence_example"}
+            Type of TFRecord protobuf being used for model training
+        feature_layer_keys_to_fns : dict
+            Dictionary of custom feature transformation functions to be applied
+            on the input features
+        max_sequence_size : int, optional
+            Maximum size of the sequence in SequenceExample protobuf
+        file_io : FileIO object
+            `FileIO` object that handles read write operations
         """
-        train_features, metadata_features = self.feature_layer_op(inputs)
-        train_features, metadata_features = self.transform_features_op(
-            train_features, metadata_features
-        )
-        return train_features, metadata_features
+        super().__init__(**kwargs)
 
-    def feature_layer_op(self, inputs: Dict[str, Input]):
-        """
-        Convert input tensorflow features into numeric
-        train features and metadata features by applying respective feature
-        transformation functions as specified in the FeatureConfig
+        self.feature_config = feature_config
+        self.tfrecord_type = tfrecord_type
+        self.max_sequence_size = max_sequence_size
 
-        Parameters
-        ----------
-        inputs : dict
-            Dictionary of the inputs to the tensorflow keras model
+        self.file_io = file_io
+
+        self.feature_layer_op = FeatureLayer(
+            feature_config=self.feature_config,
+            tfrecord_type=self.tfrecord_type,
+            feature_layer_map=self.feature_layer_map
+            file_io=self.file_io,
+            **kwargs)
+
+        self.feature_config = feature_config
+        self.tfrecord_type = tfrecord_type
+        self.feature_layer_map = feature_layer_map
+        self.file_io = file_io
+        self.all_features = self.feature_config.get_all_features(include_label=False)
 
-        Returns
-        -------
-        train_features : dict
-            Dict of feature tensors that are used for training
-        metadata_features : dict
-            Dictionary of feature tensors that can be used for
-            computing custom metrics and losses
-        """
-        raise NotImplementedError
 
-    def transform_features_op(self, train_features, metadata_features):
+class UnivariateInteractionModel(InteractionModel):
```

Comment:
I see a duplicate below. Maybe you intended to inherit from Layer in the class below

Reply:
I just noticed that the InteractionModel inherits from Layer. Any reason why the class has been split up?

Reply:
Yes, I've cleaned this up in the follow up PR. Will move the change to this.